### PR TITLE
AArch64: Add ARM64RelocatableImmInstruction

### PR DESCRIPTION
**Pull request by knn-k**
This commit adds ARM64RelocatableImmInstruction.

Signed-off-by: KONNO Kazuhiro <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
`generateSwitchToInterpreterPrePrologue()` will require the new `ARM64ImmInstruction` constructors with `RelocationType` and `SymbolReference`.

**Comment by knn-k** [comment]
@genie-omr build aarch64

File: compiler/aarch64/codegen/ARM64Instruction.hpp [hunk 1]
```diff
@@ -106,6 +106,8 @@ namespace TR
 class ARM64ImmInstruction : public TR::Instruction
    {
    uint32_t _sourceImmediate;
+   TR_ExternalRelocationTargetKind _reloKind;
+   TR::SymbolReference *_symbolReference;
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/ARM64Instruction.hpp:110]: How is an `ARM64ImmInstruction` with a SymbolReference different than an `ARM64ImmSymInstruction`?  As a developer, how would I know which form to pick?

Comment (knn-k) [reply compiler/aarch64/codegen/ARM64Instruction.hpp:110]: The main difference from `ARM64ImmSymInstruction` is `TR_ExternalRelocationTargetKind`.
Power uses the `ImmInstruction` with `TR_ExternalRelocationTargetKind` in the following lines.  The first one is without `SymbolReference`, and the other is with `SymbolReference`:
https://github.com/eclipse/omr/blob/f2285166b44222b41ecb7a73422180518a290028/compiler/p/codegen/OMRCodeGenerator.cpp#L409-L410
https://github.com/eclipse/omr/blob/f2285166b44222b41ecb7a73422180518a290028/compiler/p/codegen/OMRCodeGenerator.cpp#L425-L427

Comment (knn-k) [reply compiler/aarch64/codegen/ARM64Instruction.hpp:110]: How about making a new class `ARM64RelocatableImmInstruction`, to make the difference clearer from `ARM64ImmInstruction` and `ARM64ImmSymInstruction`?

**Comment by 0xdaryl** [comment]
Why does this extend the Instruction base class directly as opposed to extending the ImmSym instruction ?  I'm not saying that's wrong per se, but just asking the question.

**Comment by knn-k** [comment]
Because this class does not use `TR::Snippet` required by `ARM64ImmSymInstruction`.

**Comment by 0xdaryl** [comment]
@genie-omr build aarch64

**Comment by 0xdaryl** [comment]
Sorry, I missed the fact that we had discussed this already a few weeks back.

**Pull request merged**